crates/rollup-interface/src/state_machine/stf/batch.rs

```diff
@@ -0,0 +1,93 @@
+//! Transaction batch types and utilities.
+//!
+//! This module defines the core transaction types used throughout the Sovereign SDK,
+//! including fully authenticated transactions ready for DA layer submission.
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+
+use crate::Bytes;
+
+/// `FullyBakedTx` represents a serialized signed rollup transaction that has been encoded with
+/// authentication information and is ready to be placed on the DA layer.
+#[derive(
+    PartialEq,
+    Eq,
+    Clone,
+    BorshDeserialize,
+    BorshSerialize,
+    Serialize,
+    Deserialize,
+    derive_more::AsRef,
+)]
+pub struct FullyBakedTx {
+    /// Serialized transaction.
+    #[as_ref(forward)]
+    pub data: Vec<u8>,
+    /// Sequencer-provided metadata for each transaction (e.g., timestamps).
+    /// This data is NOT signed by users but is added by the sequencer.
+    pub sequencing_data: Option<Vec<u8>>,
+}
```

Comment:
As discussed in the daily, let’s use Bytes.
The following settings should enable the use of Bytes in the rollup interface.
```
borsh = { workspace = true, features = ["bytes"] }
bytes = { workspace = true, features = ["serde"] }
```